src/packagedcode/golang.py

```diff
@@ -84,6 +85,11 @@ def recognize(cls, location):
             gomod_obj = GoMod()
             gomod_data = gomod_obj.parse_gomod(location)
             yield build_gomod_package(gomod_data)
+        
+        if fileutils.file_name(location).lower() == 'go.sum':
```

Comment:
elif may be?

Reply:
Please ignore this for now. I have just created a simple initial file to get review why it is not working :)